src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala

```diff
@@ -0,0 +1,451 @@
+        }
+
+      // NOTE: Don't use DBIO.seq for **transforming** sequences
+      // - DBIO.seq(mySeq: _*) runs *any* items in sequence, but converts Seq[ DBIOAction[_] ] to DBIOAction[ Unit ]
+      // - DBIO.sequence(mySeq) converts Seq[ DBIOAction[R] ] to DBIOAction[ Seq[R] ]
+      // - DBIO.fold(mySeq, init) converts Seq[ DBIOAction[R] ] to DBIOAction[R]
+
+      _ <- DBIO.sequence(calls.toSeq map {
+        call => for {
+          executionInsert <- dataAccess.executionsAutoInc +=
+            new Execution(
+              workflowExecutionInsert.workflowExecutionId.get,
+              call.taskFqn,
+              ExecutionStatus.NotStarted.toString)
+
+          _ <- backend match {
+            case _: LocalBackend =>
+              dataAccess.localJobsAutoInc +=
+                new LocalJob(
+                  executionInsert.executionId.get,
+                  -1,
+                  -1)
+            case null =>
+              throw new NotImplementedError("Backend is null")
+            case unknown =>
+              throw new NotImplementedError("Unknown backend: " + backend.getClass)
+          }
+        } yield ()
+      })
+
+    } yield ()
+
+    runTransaction(action)
+  }
+
+  override def getWorkflowState(workflowId: WorkflowId): Future[Option[WorkflowState]] = {
+
+    val action = for {
+
+      workflowExecutionResultOption <- dataAccess.workflowExecutions.filter(
+        _.workflowExecutionUuid === workflowId.toString).result.headOption
+
+      workflowState = workflowExecutionResultOption map { workflowExecutionResult =>
+        WorkflowState.fromString(workflowExecutionResult.status)
+      }
+
+    } yield workflowState
+
+    runTransaction(action)
+  }
+
+  override def getExecutionStatuses(workflowId: WorkflowId): Future[Map[FullyQualifiedName, CallStatus]] = {
+
+    val action = for {
+
+      workflowExecutionResult <- dataAccess.workflowExecutions.filter(
+        _.workflowExecutionUuid === workflowId.toString).result.head
+
+      executionResults <- dataAccess.executions.filter(symbol =>
+        symbol.workflowExecutionId === workflowExecutionResult.workflowExecutionId).result
+
+      executionStatuses = executionResults.map(executionResult =>
+        executionResult.callFqn -> ExecutionStatus.withName(executionResult.status)
+      ).toMap
+
+    } yield executionStatuses
+
+    runTransaction(action)
+  }
+
+  override def getWorkflowsByState(states: Traversable[WorkflowState]): Future[Traversable[WorkflowInfo]] = {
+
+    val action = for {
+
+      workflowExecutionResults <- dataAccess.workflowExecutions.filter(
+        _.status inSet states.map(_.toString)).result
+
+      workflowInfos <- DBIO.sequence(
+        workflowExecutionResults map { workflowExecutionResult =>
+          val workflowExecutionAuxResult = dataAccess.workflowExecutionAuxes.filter(
+            _.workflowExecutionId === workflowExecutionResult.workflowExecutionId).result.head
```

Comment:
Yep, I'm not a db guru, but if we ever need to merge the two tables, they should be as Miguel described. Will return a `Failure` if not.